docs/tutorials/local_iteration_series/testing_with_artificial_conditions.md

```diff
@@ -0,0 +1,173 @@
+---
+id: testing_with_artificial_conditions
+title: Local Iteration - Testing multiplayer games with artificial network conditions
+description: Guide covering the available tools for local testing of multiplayer games with artificial latency, jitter and packet loss .
+---
+- [General advice](#general-advice)
+	- [Different network conditions for different peers](#different-network-conditions-for-different-peers)
+- [Unity Transport - Simulator Tools](#unity-transport---simulator-tools)
+	- [Clone-based workflow (ParrelSync)](#clone-based-workflow-parrelsync)
+	- [Debug Builds](#debug-builds)
+- [System-wide network conditioners](#system-wide-network-conditioners)
+	- [Clumsy on Windows](#clumsy-on-windows)
+		- [Settings quickstart](#settings-quickstart)
+	- [Network Link Conditioner on OSX](#network-link-conditioner-on-osx)
+		- [Settings quickstart](#settings-quickstart-1)
+
+
+When we are developing a multiplayer game that is intended to operate over the internet we inevitably have to deal with the reality of poor network conditions.
+
+Adverse network factors, such as [latency, jitter and packet loss](lagandpacketloss.md) all have to be taken into account during development and the tolerance thresholds for these should not be an afterthought or something relegated to an "optimization pass". 
+
+:::warning
+
+Not testing with artificial network conditions makes it significantly more likely to have unexpected behaviors when the game is running over the internet.
+
+:::
+
+This is challenging when we are [iterating on our game locally](local-Iteration-testing-multiplayer-games-locally.md) - after all, all of the game instances are still running on the same network interface. It's reasonable to expect that there will be little to no latency between the clients, which is no good for our purposes - we want the network to **misbehave**. 
+
+Thankfully there are a number of tools that can simulate adverse network conditions.
+
+For testing locally within the editor we suggest using [Unity Transport - Simulator Tools](#unity-transport---simulator-tools) along with [clone-based workflow via ParrelSync](testing_locally.md#parrelsync).
+
+For testing development builds with built-in artificial latency we suggest using [Unity Transport Simulator Tools with some custom code to inject artifical conditions into the build](#debug-builds).
+
+For testing release builds we suggest using either [Clumsy](#clumsy-on-windows) if you're on Windows, and [Network Link Conditioner](#network-link-conditioner-on-osx) if you're on Mac OS.
```

Comment:
mmm so it might be worth it to have suggestions for other platforms too. Or at least mention that most platforms have their own way to simulate artificial latency. iOS has dev tools for this, I'm sure android and consoles do too. 
It could be an addition later too, once we actually get to test this ourselves.

Reply:
Yeah I thought about mobile platforms and even consoles, but until we can test it - feels weird to recommend solutions until we test them ourselves. I can add a general statement that be on the lookout for platform-specific tools if you are working on a platform other than Win / Mac.

Reply:
good idea
For testing, we'll need to look into dev consoles very soon.
For mobile, in theory we should already have what it takes. I have iOS devices and @fernando-cortez has Android ones. I can reserve some time next week to test this if you want

Reply:
I would expect we cover recommendations for the platforms we say we support. I understand the difficulties for consoles while we don't have dev kits. I suggest reaching out to the console support teams lead by Jason Page for knowledge and guidance.

Reply:
So it's important that we remember that this document is live, we can add to it. I don't want to hold up its release. My question here is would this be releasable with its current content or would it need more content on the other platforms for launch.

Reply:
As I said, at the least the platforms that NGO supports.

Reply:
Reached out to Jason Page regarding touching on the tools for consoles.
We had a small discussion with Sam and the current idea is:
 - document Unity's stuff (UTP + Parrelsync for now)
 - touch on some tools that are easy to touch on (osx and windows stuff)
 - mention the rest (NetworkLinkConditioner on iOS, console stuff, Android and so forth).

Reply:
we want to avoid documenting the tools that other companies develop too heavily because it's a pain to keep these up to date. We would let people know that tools for their platforms exist and refer them to the docs.

Reply:
Yes, I meant to elaborate further; I certainly didn't mean we document the individual tools Unity are not responsible for developing but I wanted to see tools called out on a per platform basis. If we can link to them as well, great, otherwise just bring awareness is suitable.

Reply:
Overwatch fails elegantly above 300ms. In the sense where you can still play the game, but server rewind will stop helping you after 300ms and so you'll start missing your shots more. This is a choice Blizzard made. Other FPS devs will not put such a restriction and will allow 500ms and even 1000ms latency.
In past projects, our QA team would test with 100ms latency, 200ms, 500ms and 1000ms and give a report on playability. It'd then be up to designers/leads to determine if the issues found were acceptable or not, if they were ship blockers or not.
Determining which network conditions to test with is NOT about what type of game you have. It's about what your users will encounter while playing your game, which will vary by region and by platform, not by genre.
Saying an FPS will require lower latencies doesn't make sense --> tell that to your roommate trying to download game of thrones while you play. We have lag hiding techniques for this, we have prediction, we have server rewind, we have interpolation/buffering to hide that jitter, but that 200ms of lag won't disappear, it's completely our of your control as a player and as a dev. It's up to the dev to make sure it doesn't show. And to do that, you need to test with realistic network conditions and use the right techniques to hide it successfully.

Now determining what bugs are ACCEPTABLE after you tests latency, that's dependent on game genre. If I have a turn by turn card game and I don't get an update after 500ms due to latency, I wouldn't mind. Or if my tank is moving slow because I'm not predicting it, I wouldn't mind (I'd call this a feature and say it's due to the big thing's intertia... :P ) But you should still test at 100ms, 200ms, 500ms, 1000ms, to make sure you don't have weird race conditions, since your users WILL have those latencies.

Again, test values for artificial network conditions are orthogonal to game genre.